src/components/Tokens/TokenDetails/StatsSection.tsx

```diff
@@ -77,44 +83,100 @@ function Stat({
 type StatsSectionProps = {
   chainId: ChainId
   address: string
-  priceLow52W?: NumericStat
-  priceHigh52W?: NumericStat
-  TVL?: NumericStat
-  volume24H?: NumericStat
+  tokenQueryData: TokenQueryData
 }
 export default function StatsSection(props: StatsSectionProps) {
-  const { chainId, address, priceLow52W, priceHigh52W, TVL, volume24H } = props
+  const { chainId, address, tokenQueryData } = props
   const { label, infoLink } = getChainInfo(chainId)
+  const isInfoTDPEnabled = useInfoTDPEnabled()
+
+  const tokenMarketInfo = tokenQueryData?.market
+  const tokenProjectMarketInfo = tokenQueryData?.project?.markets?.[0] // aggregated market price from CoinGecko
+
+  const FDV = tokenProjectMarketInfo?.fullyDilutedValuation?.value
+  const marketCap = tokenProjectMarketInfo?.marketCap?.value
+  const TVL = tokenMarketInfo?.totalValueLocked?.value
+  const volume24H = tokenMarketInfo?.volume24H?.value
+  const priceHigh52W = tokenMarketInfo?.priceHigh52W?.value
+  const priceLow52W = tokenMarketInfo?.priceLow52W?.value
+
+  const haveStats = isInfoTDPEnabled
+    ? TVL || FDV || marketCap || volume24H
+    : TVL || volume24H || priceLow52W || priceHigh52W
 
-  if (TVL || volume24H || priceLow52W || priceHigh52W) {
+  if (haveStats) {
     return (
       <StatsWrapper data-testid="token-details-stats">
         <Header>
           <Trans>Stats</Trans>
         </Header>
         <TokenStatsSection>
-          <StatPair>
-            <Stat
-              dataCy="tvl"
-              value={TVL}
-              description={HEADER_DESCRIPTIONS[TokenSortMethod.TOTAL_VALUE_LOCKED]}
-              title={<Trans>TVL</Trans>}
-            />
-            <Stat
-              dataCy="volume-24h"
-              value={volume24H}
-              description={
-                <Trans>
-                  24H volume is the amount of the asset that has been traded on Uniswap v3 during the past 24 hours.
-                </Trans>
-              }
-              title={<Trans>24H volume</Trans>}
-            />
-          </StatPair>
-          <StatPair>
-            <Stat dataCy="52w-low" value={priceLow52W} title={<Trans>52W low</Trans>} />
-            <Stat dataCy="52w-high" value={priceHigh52W} title={<Trans>52W high</Trans>} />
-          </StatPair>
+          {isInfoTDPEnabled ? (
+            <>
+              <StatPair>
+                <Stat
+                  dataCy="tvl"
+                  value={TVL}
+                  description={HEADER_DESCRIPTIONS[TokenSortMethod.TOTAL_VALUE_LOCKED]}
+                  title={<Trans>TVL</Trans>}
+                />
+                <Stat
+                  dataCy="market-cap"
+                  value={marketCap}
+                  description={
+                    <Trans>
+                      Market capitalization is the total market value of an asset&apos;s circulating supply.
+                    </Trans>
```

Comment:
should we add all descriptions to the `HEADER_DESCRIPTIONS` const?

Reply:
no - only the variables that will be in the token table